Source/FieldSolver/WarpX_QED_K.H

```diff
@@ -0,0 +1,278 @@
+#ifndef WarpX_QED_K_h
+#define WarpX_QED_K_h
+
+#include <AMReX_FArrayBox.H>
+#include <WarpXConst.H>
+#include <cmath>
+
+using namespace amrex;
+
+/**
+ *warpx_hybrid_QED_push uses an FDTD scheme to calculate QED corrections to Maxwell's equations and
+ *preforms a half timestep correction to the E-fields
+ *\param[in] Ex This function modifies the Ex field at the end
+ *\param[in] Ey This function modifies the Ey field at the end
+ *\param[in] Ez This function modifies the Ez field at the end
+ *\param[in] Bx The QED corrections are non-linear functions of B. However,
+ *they do not modify B itself
+ *\param[in] By The QED corrections are non-linear functions of B. However,
+ *they do not modify B itself
+ *\param[in] Bz The QED corrections are non-linear functions of B. However,
+ * they do not modify B itself
+ *\param[in] tempEx Since the corrections to E at a given node are non-linear functions
+ *of the values of E on the surronding nodes, temp arrays are used so that modifications to
+ *one node do not influence the corrections to the surronding nodes
+ *\param[in] tempEy Since the corrections to E at a given node are non-linear functions
+ *of the values of E on the surronding nodes, temp arrays are used so that modifications to
+ *one node do not influence the corrections to the surronding nodes
+ *\param[in] tempEz Since the corrections to E at a given node are non-linear functions
+ * of the values of E on the surronding nodes, temp arrays are used so that modifications to
+ *one node do not influence the corrections to the surronding nodes
+ *\param[in] dx The x spatial step, used for calculating curls
+ *\param[in] dy The y spatial step, used for calculating curls
+ *\param[in] dz The z spatial step, used for calulating curls
+ *\param[in] dt The temporal step, used for the half push/correction to the E-fields at the end of the function
+ */
+
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+
+void calc_M(Real arr [], Real ex, Real ey, Real ez, Real bx, Real by, Real bz, Real xi, Real c2)
```

Comment:
Just reformatting?
```
/**
 * warpx_hybrid_QED_push uses an FDTD scheme to calculate QED corrections to
 * Maxwell's equations and preforms a half timestep correction to the E-fields
 *
 * \param[in] Ex This function modifies the Ex field at the end
 * \param[in] Ey This function modifies the Ey field at the end
 * \param[in] Ez This function modifies the Ez field at the end
 * \param[in] Bx The QED corrections are non-linear functions of B. However,
 *            they do not modify B itself
 * \param[in] By The QED corrections are non-linear functions of B. However,
 *            they do not modify B itself
 * \param[in] Bz The QED corrections are non-linear functions of B. However,
 *            they do not modify B itself
 * \param[in] tempEx Since the corrections to E at a given node are non-linear functions
 *            of the values of E on the surronding nodes, temp arrays are used so that
 *            modifications to one node do not influence the corrections to the surronding nodes
 * \param[in] tempEy Since the corrections to E at a given node are non-linear functions
 *            of the values of E on the surronding nodes, temp arrays are used so that modifications to
 *            one node do not influence the corrections to the surronding nodes
 * \param[in] tempEz Since the corrections to E at a given node are non-linear functions
 *            of the values of E on the surronding nodes, temp arrays are used so that modifications to
 *            one node do not influence the corrections to the surronding nodes
 * \param[in] dx The x spatial step, used for calculating curls
 * \param[in] dy The y spatial step, used for calculating curls
 * \param[in] dz The z spatial step, used for calulating curls
 * \param[in] dt The temporal step, used for the half push/correction to the E-fields at the end of the function
 */
AMREX_GPU_HOST_DEVICE AMREX_INLINE 
void calc_M(Real arr [], Real ex, Real ey, Real ez, Real bx, Real by, Real bz, Real xi, Real c2)
```

Reply:
BTW isn't this description misplaced? Could you move it to the proper place (function `warpx_hybrid_QED_push`) and add a doctoring for `calc_M`?